src/t8_geometry/t8_geometry_helpers.c

```diff
@@ -411,6 +411,77 @@ t8_geom_get_scaling_factor_of_edge_on_face_tet (const int edge, const int face,
   return (1.0 - (orthogonal_vector[edge][face] / max_orthogonal_vector[edge][face]));
 }
 
+double
+t8_geom_get_scaling_factor_of_edge_on_face_pyramid (const int edge, const int face, const double *ref_coords)
+{
+  /* Save the orthogonal direction and the maximum of that direction
+   * of an edge in reference space on one of the neighbouring faces. 
+   *           /|
+   *          / |
+   *         /  |
+   *        /   |
+   *       /    |--edge
+   *      /   --|--face
+   *     /      |
+   *    /    <~~| orthogonal direction
+   *   /<----o--| maximum othogonal direction
+   *  /_________|
+   */
+  const double orthogonal_direction[8][5]
+    = { { ref_coords[2], 0, 0, 0, ref_coords[0] },
+        { 0, ref_coords[2], 0, 0, (1 - ref_coords[0]) },
+        { 0, 0, ref_coords[2], 0, ref_coords[1] },
+        { 0, 0, 0, ref_coords[2], (1 - ref_coords[1]) },
+        { (ref_coords[1] - ref_coords[2]), 0, (ref_coords[0] - ref_coords[2]), 0, 0 },
+        { 0, (ref_coords[1] - ref_coords[2]), (1 - ref_coords[0]), 0, 0 },
+        { 0, (1 - ref_coords[1]), 0, (1 - ref_coords[0]), 0 },
+        { (1 - ref_coords[1]), 0, 0, (ref_coords[0] - ref_coords[2]), 0 } };
+  const double max_orthogonal_direction[8][5] = { { ref_coords[1], 0, 0, 0, 1 },
+                                                  { 0, ref_coords[1], 0, 0, 1 },
+                                                  { 0, 0, ref_coords[0], 0, 1 },
+                                                  { 0, 0, 0, ref_coords[0], 1 },
+                                                  { (1 - ref_coords[2]), 0, (1 - ref_coords[0]), 0, 0 },
+                                                  { 0, (1 - ref_coords[2]), (1 - ref_coords[2]), 0, 0 },
+                                                  { 0, (1 - ref_coords[2]), 0, (1 - ref_coords[2]), 0 },
+                                                  { (1 - ref_coords[2]), 0, 0, (1 - ref_coords[2]), 0 } };
```

Comment:
Most of the values are calculated and not used

Reply:
Every value which is not 0 will get used at some point. Removing the 0's is not possible, since calling the lookup with the edge number wouldn't be possible anymore